### PR TITLE
Support link names in HTTP-server, refactor for flattened config

### PR DESCRIPTION
**Pull request by brooksmtownsend**
- **feat(http-server): support link names**
- **feat(http-server)!: use flattened config, document well**

## Feature or Problem
This PR provides a suite of refactors to the HTTP server to support link names and flatten configuration, still remaining backwards compatible for supplying configuration via `config_b64` or `config_json` if desired.

This also corrects a good bit of our documentation, and removes some fairly old structures. There's more to do that I plan on addressing in followups when approaching routing

### CODE REVIEW

**Review by lxfontes — APPROVED** [review]

File: crates/provider-http-server/README.md [hunk 1]
```diff
@@ -1,20 +1,38 @@
 # wasmCloud HTTP Server Provider
 
-This capability provider implements the `wasmcloud:httpserver` capability contract, and enables a component to accept incoming HTTP(s) requests. It is implemented in Rust with the [warp](https://docs.rs/warp/) web server framework and the fast and scalable [hyper](https://docs.rs/hyper/) http implementation.
+This capability provider imports the `wasi:http/incoming-handler` interface and enables a component to accept incoming HTTP(s) requests. It is implemented in Rust with the [axum](https://docs.rs/axum/) web server framework and the fast and scalable [hyper](https://docs.rs/hyper/) HTTP implementation. Its goal is to provide a multitenant, high performance HTTP server that proxies requests to components based on the configuration of the HTTP server.
```
Comment (vados-cosmonic) [inline crates/provider-http-server/README.md:3]:
```suggestion
This capability provider imports the `wasi:http/incoming-handler` interface and enables a component to accept incoming HTTP(s) requests. It is implemented in Rust with the [axum](https://docs.rs/axum/) web server framework and the fast and scalable [hyper](https://docs.rs/hyper/) HTTP implementation. 

This HTTP provider's goal is to provide a multitenant, high performance HTTP server that proxies requests to components based on the configuration of the HTTP server.
```

File: crates/provider-http-server/README.md [hunk 2]
```diff
@@ -1,20 +1,38 @@
...
+wash build -p src/bin/http-server-provider
+```
 
-The default listen address is 127.0.0.1 port 8000.
+## Provider Configuration
 
-### ⚠️ Caution - Port Ownership
-If the instance of this capability provider running on a single host is linked to multiple components attempting to claim the same port, only the first **link definition** for that port will succeed, and the subsequent attempts will fail. During development, 
-it is recommended to check ("tail") the wasmCloud host logs for success and error messages.
+The wasmCloud HTTP server has optional configuration that you can provide to it at startup. All configuration keys should be lowercased, as well as their values when applicable. See the [provider config](https://wasmcloud.com/docs/developer/providers/configure) documentation for information about defining and using this configuration.
```
Comment (vados-cosmonic) [inline crates/provider-http-server/README.md:15]:
```suggestion
The wasmCloud HTTP server has optional configuration that you can provide to it at startup. All configuration keys should be lowercased, as well as their values when applicable. See the [provider config documentation](https://wasmcloud.com/docs/developer/providers/configure) for information about defining and using this configuration.
```

File: crates/provider-http-server/README.md [hunk 3]
```diff
@@ -1,20 +1,38 @@
...
+| Key              | Value                  | Default        | Description                                                                                                                                                                                                            |
+| ---------------- | ---------------------- | -------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `routing_mode`   | "address","path"       | `address`      | Dictates the routing mode of the capability provider. `address` mode will listen on a new address for each component, and `path` mode will listen on a single address and route to components based on a provided path |
+| `listen_address` | A valid listen address | "0.0.0.0:8080" | Must be provided when using the `path` routing mode. The listen address to listen on and route to components based on path                                                                                             |
+
+<!-- TODO: consider setting this up as the default address? -->
+
```
Comment (vados-cosmonic) [inline crates/provider-http-server/README.md:23]: What is this TODO from?

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket:

File: crates/provider-http-server/README.md (hunk 3 above)
Comment (brooksmtownsend) [reply crates/provider-http-server/README.md:23]: Oh dang, this is from a future enhancement that I left on accident. Will remove

**Pull request merged**